### ISSUE TITLE
Steamshovel freezes when rotating I3Geometry with High Quality DOMs (Trac #503)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#502]
In Ubuntu 13.04, running steamshovel with I3Geometry loaded, if I try to rotate the geometry then steamshovel will freeze and nothing will happen if the option "High Quality DOMs" is set. I have no problems rotating any other object. If I turn off the geometry and rotate the other objects, eventually I can turn the I3Geometry back on and it will be possible to rotate it again without freezing. 

Migrated from https://code.icecube.wisc.edu/ticket/503
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:42", 
    "description": "In Ubuntu 13.04, running steamshovel with I3Geometry loaded, if I try to rotate the geometry then steamshovel will freeze and nothing will happen if the option \"High Quality DOMs\" is set. I have no problems rotating any other object. If I turn off the geometry and rotate the other objects, eventually I can turn the I3Geometry back on and it will be possible to rotate it again without freezing. ", 
    "reporter": "icecube", 
    "cc": "david.schultz@icecube.wisc.edu", 
    "resolution": "fixed", 
    "_ts": "1550067102580394", 
    "component": "combo core", 
    "summary": "Steamshovel freezes when rotating I3Geometry with High Quality DOMs", 
    "priority": "major", 
    "keywords": "steamshovel", 
    "time": "2014-04-23T21:17:27", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20140423T22:28:54: david.schultz changed owner from "sjackso" to "david.schultz"

**Comment by icecube-trac** [comment]
Trac update at 20141017T11:34:06: boersma commented:

> When I start steamshovel, the first thing I do is: click in the event display window and type the 1 (the digit one) key once. This is an obscure trick that I learned from sjackso long ago. I guess someone could dive into the code and check what the effect of this key stroke actually is (maybe it is the high quality dom setting), but for me it helps to get a relatively normal steamshovel experience. If I forget about the 1 key voodoo, then I get problems similar to described in this ticket. It would be nice if this setting were "remembered" between sessions so that I do not really have to do it *every* time.

**Comment by icecube-trac** [comment]
Trac update at 20150323T22:52:55: olivas changed component from "steamshovel" to "icecube offline"

**Comment by icecube-trac** [comment]
Trac update at 20150709T22:17:38: 
* david.schultz changed milestone from "" to "offline V15-07-00"
* david.schultz changed status from "new" to "assigned"
* david.schultz commented:

> We should be able to time the first draw with high quality doms, and automatically go to lower quality if that is really slow.
> 
> Assigning to Hans so he sees this ticket, since I can't cc him.


* david.schultz changed owner from "david.schultz" to "hdembinski"
* david.schultz changed cc from "" to "david.schultz@icecube.wisc.edu"

**Comment by icecube-trac** [comment]
Trac update at 20150710T14:04:52: hdembinski commented:

> Hm, it works for me, I always run with high quality doms. A slow computer perhaps? Is this still valid?

**Comment by icecube-trac** [comment]
Trac update at 20150710T15:03:43: rcross commented:

> I get this bug when I run steamshovel on my integrated intel chip. It works fine when I use my nvidia chip. Maybe the intel driver doesn't like the sphere_matte.frag GLSL shader?
> 
> 
> '''Intel'''
> 
> ```text
> steamshovel » glxinfo | grep version
> server glx version string: 1.4
> client glx version string: 1.4
> GLX version: 1.4
> OpenGL core profile version string: 3.3 (Core Profile) Mesa 10.5.8
> OpenGL core profile shading language version string: 3.30
> OpenGL version string: 3.0 Mesa 10.5.8
> OpenGL shading language version string: 1.30
> OpenGL ES profile version string: OpenGL ES 3.0 Mesa 10.5.8
> OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.00
> 
> steamshovel » glxinfo | grep shading
> OpenGL core profile shading language version string: 3.30
>     GL_ARB_provoking_vertex, GL_ARB_robustness, GL_ARB_sample_shading, 
>     GL_ARB_shading_language_420pack, GL_ARB_shading_language_packing, 
> OpenGL shading language version string: 1.30
>     GL_ARB_robustness, GL_ARB_sample_shading, GL_ARB_sampler_objects, 
>     GL_ARB_shader_texture_lod, GL_ARB_shading_language_100, 
>     GL_ARB_shading_language_420pack, GL_ARB_shading_language_packing, 
> OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.00
> ```
> 
> '''Nvidia'''
> 
> 
> ```text
> steamshovel » primusrun glxinfo | grep version
> server glx version string: 1.4
> client glx version string: 1.4
> GLX version: 1.4
> OpenGL core profile version string: 4.4.0 NVIDIA 340.76
> OpenGL core profile shading language version string: 4.40 NVIDIA via Cg compiler
> OpenGL version string: 4.4.0 NVIDIA 340.76
> OpenGL shading language version string: 4.40 NVIDIA via Cg compiler
> 
> steamshovel » primusrun glxinfo | grep shading
> OpenGL core profile shading language version string: 4.40 NVIDIA via Cg compiler
>     GL_ARB_robustness, GL_ARB_sample_shading, GL_ARB_sampler_objects, 
>     GL_ARB_shader_texture_lod, GL_ARB_shading_language_100, 
>     GL_ARB_shading_language_420pack, GL_ARB_shading_language_include, 
>     GL_ARB_shading_language_packing, GL_ARB_shadow, GL_ARB_sparse_texture, 
> OpenGL shading language version string: 4.40 NVIDIA via Cg compiler
>     GL_ARB_robustness, GL_ARB_sample_shading, GL_ARB_sampler_objects, 
>     GL_ARB_shader_texture_lod, GL_ARB_shading_language_100, 
>     GL_ARB_shading_language_420pack, GL_ARB_shading_language_include, 
>     GL_ARB_shading_language_packing, GL_ARB_shadow, GL_ARB_sparse_texture,
> ```

**Comment by icecube-trac** [comment]
Trac update at 20150724T16:28:07: 
* david.schultz commented:

> I can reproduce the problem on my laptop.  Interestingly, disabling the shader, rotating the geometry slightly, and reenabling the shader makes it work just fine.  Probably a loading issue?
> 
> More debugging is needed, but ran out of time.


* david.schultz changed milestone from "offline V15-07-00" to "offline V15-09-00"
* david.schultz changed status from "assigned" to "accepted"
* david.schultz changed owner from "hdembinski" to "david.schultz"

**Comment by icecube-trac** [comment]
Trac update at 20160127T21:05:37: 
* david.schultz changed priority from "normal" to "major"
* david.schultz commented:

> Noting that the same issue and "fix" still exists.


* david.schultz changed milestone from "offline V15-12-00" to "April 2016"

**Comment by icecube-trac** [comment]
Trac update at 20160129T19:55:16: david.schultz commented:

> The actual issue is narrowed down to `Scene::pick`. The Intel driver doesn't like drawing on a fake buffer or something. Disabling the shader removes most of the issue, but there is still a 0.3 second drawing time, which will cause lag whenever the mouse button is clicked. That will be tackled in #1527.

**Comment by icecube-trac** [comment]
Trac update at 20160129T19:56:31: 
* david.schultz changed status from "accepted" to "closed"
* david.schultz changed resolution from "" to "fixed"
* david.schultz commented:

> In changeset 141356:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="141356"
> disable the shader in Scene::pick to speed up mouse clicks on Intel drivers. closes #502
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:42: 
* olivas changed milestone from "April 2016" to ""
* olivas commented:

> Milestone April 2016 deleted
